### DISCUSSION TITLE
Is it possible to route to device

### DISCUSSION THREAD

**Discussion by KemalErtas** [anywherelan/awl#101] [General]
![245154095-9ff299dc-5946-4604-8a28-98cca10c2730](https://github.com/anywherelan/awl/assets/109876032/31cc8fce-40c8-486b-a526-94d460e97db6)

**Comment by pymq** [comment]
Unfortunately, not at this time. Awl is not compatible with system-level packets routing. Currently awl implements vpn like a NAT, it rewrites src and dsc addresses in all packets.

What do you want to achieve with this setup? You want to have access from one network to another using awl (setting awl address as a default route)?
I have plans to support this setup out of the box using only awl, but it will take time to implement.
For now I can only suggest you to install awl on each device you want to have access to. This way you could have an isolated network between these devices (peers).

**Reply by KemalErtas** [reply]
Device is a PLC, ROBOT, motor speed drive etc. 
So its impossible as a peer.

**Reply by pymq** [reply]
Ah, I see. But you can install awl on some device in that network. And have access to your PLC from that device. Not that convenient of course

**Reply by KemalErtas** [reply]
In my PC, there is PLC programmimg tool.
On remote site, there is a PC with SCADA and PLC connected by ethernet.
I want to monitor and/or modify PLC programs on PLC by using my programmimg tool on my PC.

**Reply by pymq** [reply]
As I understand you can just connect to 10.66.0.2 (peer 2) from peer 1 using RDP or VNC or SSH for example? And communicate with PLC from peer 2 as usual

**Reply by KemalErtas** [reply]
Yes

**Reply by KemalErtas** [reply]
Sorry not yes.
Peer 1 Will communicate with peer 2 using Vpn. Peer 2 Will be used as a router to PLC.
I will ping PLC from my PC (peer 1). 
My PC (peer1)Will route 192.168.2.0 frames by using 10.66.0.2
Peer2 has ethernet with 192.168.2.1. Peer2 Will route 192.168.2.0 frames to PLC with 192.168.2.10

**Reply by pymq** [reply]
What protocol does your PLC use for communication? is it something custom over TCP/UDP? In that case I could recommend you to use [SSH port forwarding](https://linuxize.com/post/how-to-setup-ssh-tunneling/) from your Peer2 (which has connection to PLC) to Peer1.

So it would be like:
Peer1 -> Peer2 by awl
Peer2 -> PLC by ethernet (has direct access)
Peer1 -> PLC by SSH port forwarding

Yeah, this is a bit of a complicated setup, nothing better comes to mind, I'm sorry

**Reply by KemalErtas** [reply]
its possible with teamviewer VPN.

On peer 2 (192.168.2.1), first I enabled the  "ip routing on registery" with win10 regedit.

_PLC IP setting : 192.168.2.10
mask 255.255.255.0
Gateway : 192.168.2.1_

After VPN connection , just writing a command on cmd window on my PC (peer1)  _"route add 192.168.2.0 mask 255.255.255.0  10.66.0.2 metric 1"_
assuming  peer2 VPN IP is 10.66.0.2

Then I can ping PLC from my PC.
